src/VisualStudio/Core/Next/Remote/ServiceHubRemoteHostClient.cs

```diff
@@ -42,8 +42,8 @@ internal sealed partial class ServiceHubRemoteHostClient : RemoteHostClient
                 // Retry (with timeout) until we can connect to RemoteHost (service hub process). 
                 // we are seeing cases where we failed to connect to service hub process when a machine is under heavy load.
                 // (see https://devdiv.visualstudio.com/DevDiv/_workitems/edit/481103 as one of example)
-                var instance = await RetryRemoteCallAsync<(Workspace workspace, HubClient primary, TimeSpan timeout), IOException, ServiceHubRemoteHostClient>(
-                    CreateWorkerAsync, (workspace, primary, timeout), timeout, cancellationToken).ConfigureAwait(false);
+                var instance = await RetryRemoteCallAsync<IOException, ServiceHubRemoteHostClient>(
```

Comment:
is the second type parameter necessary?  can it not be inferred now from the delegate passed in?

Reply:
oh.  you need both, since you have to specify the exception type.